benchmarks/benchmark_rank.py

```diff
@@ -2,17 +2,24 @@
 from skimage import img_as_ubyte
 from skimage.filters import rank
 from skimage.filters.rank import __all__ as all_rank_filters
-from skimage.morphology import grey, disk
+from skimage.filters.rank import __3Dfilters__ as all_3d_rank_filters
+from skimage.morphology import grey, disk, ball
 
 
 class RankSuite(object):
 
-    param_names = ["filter", "shape"]
-    params = [sorted(all_rank_filters), [(32, 32), (256, 256)]]
+    param_names = ["filter", "shape", "filter3d", "shape3d"]
+    params = [sorted(all_rank_filters), [(32, 32), (256, 256)],
+              sorted(all_3d_rank_filters), [(32, 32, 32), (256, 256, 256)]]
```

Comment:
I would just split the 2D and 3D cases to separate classes. Otherwise, the way this is now there all 2D cases get repeated for every 3D case and vice versa
.
In other words set of `params` results in `len(all_rank_filters) * 2 * len(all_3d_rank_filters) * 2` test cases instead of  the intended `len(all_rank_filters) * 2 + len(all_3d_rank_filters) * 2` cases

e.g. the output looks like...
```
             ------------------------------------------------------------- --------------
                          filter               shape          filter3d       (32, 32, 32) 
              ============================= ============ ================== ==============
                        autolevel             (32, 32)       autolevel         33.0±0ms   
                        autolevel             (32, 32)    enhance_contrast     21.2±0ms   
                        autolevel             (32, 32)        entropy          41.7±0ms   
                        autolevel             (32, 32)        equalize         32.3±0ms   
                        autolevel             (32, 32)     geometric_mean      37.4±0ms   
                        autolevel             (32, 32)        gradient         34.4±0ms   
                        autolevel             (32, 32)        majority         26.4±0ms   
                        autolevel             (32, 32)        maximum          32.7±0ms   
                        autolevel             (32, 32)          mean           35.2±0ms   
                        autolevel             (32, 32)         median          22.5±0ms   
                        autolevel             (32, 32)        minimum          32.1±0ms   
                        autolevel             (32, 32)         modal           36.6±0ms   
                        autolevel             (32, 32)      noise_filter       35.6±0ms   
                        autolevel             (32, 32)          otsu           42.2±0ms   
                        autolevel             (32, 32)          pop            30.6±0ms   
                        autolevel             (32, 32)     subtract_mean       36.8±0ms   
                        autolevel             (32, 32)          sum            23.5±0ms   
                        autolevel             (32, 32)       threshold         37.7±0ms   
                        autolevel            (256, 256)      autolevel         21.2±0ms   
                        autolevel            (256, 256)   enhance_contrast     32.4±0ms   
                        autolevel            (256, 256)       entropy          44.2±0ms   
                        autolevel            (256, 256)       equalize         30.6±0ms   
                        autolevel            (256, 256)    geometric_mean      40.5±0ms   
                        autolevel            (256, 256)       gradient         29.8±0ms   
                        autolevel            (256, 256)       majority         38.9±0ms   
                        autolevel            (256, 256)       maximum          20.8±0ms   
                        autolevel            (256, 256)         mean           35.1±0ms   
                        autolevel            (256, 256)        median          34.7±0ms   
                        autolevel            (256, 256)       minimum          29.1±0ms   
                        autolevel            (256, 256)        modal           39.5±0ms   
                        autolevel            (256, 256)     noise_filter       23.6±0ms   
                        autolevel            (256, 256)         otsu           42.5±0ms   
                        autolevel            (256, 256)         pop            31.1±0ms   
                        autolevel            (256, 256)    subtract_mean       35.3±0ms   
                        autolevel            (256, 256)         sum            36.8±0ms   
                        autolevel            (256, 256)      threshold         35.3±0ms   
```

Reply:
also (256, 256, 256) takes pretty long to run (~10 s for a single run for each filter), so perhaps reduce to (128, 128, 128) to make the overall run time more reasonable.